UnityProject/Assets/Scripts/Lighting/LightSource.cs

```diff
@@ -97,9 +99,16 @@ private void OnStateChange(LightState iValue)
 		private void Awake()
 		{
 			Renderer = GetComponentInChildren<SpriteRenderer>();
-
-			// Slight color variance.
-			Color _color = new Color(0.7264151f, 0.7264151f, 0.7264151f, 0.8f); //+ UnityEngine.Random.ColorHSV() * 0.3f;
+			Color _color;
+			
+			if (customColor == new Color(0, 0, 0, 0))
+			{
+			_color = new Color(0.7264151f, 0.7264151f, 0.7264151f, 0.8f);
+			}
+			else
+			{
+				_color = customColor;
+			}
 
 			mLightRendererObject = LightSpriteBuilder.BuildDefault(gameObject, _color, 12);
```

Comment:
You need to check if mLightRendererObject is null here before building a light sprite for it. That is because a custom light sprite can be mapped via the editor and referenced in the mLightRendererObject field via the inspector.

Reply:
alrighty managed to fix it for you. Will merge now